docs/development/sync-protocol.md

```diff
@@ -0,0 +1,431 @@
+
+##### `eventupload` Response Example Payload 
+
+
+```json
+{
+   "event_upload_bundle_binaries": ["8621d92262aef379d3cfe9e099f287be5b996a281995b5cc64932f7d62f3dc85"]
+}
+```
+
+### Rule Download
+
+After Events have been uploaded to the Sync server, the Client then initiates the `rulesdownload` request. 
+
+Like the previous stages this is a simple HTTP request response cycle like so:
+
+```mermaid
+sequenceDiagram
+   client ->> server: POST /ruledownload/<machine_id>
+   server -->> client: ruledownload response
+```
+
+This stage may be performed many times depending on the rule batch size set during the `preflight` stage. 
+
+In the event that the client requested a clean sync or the server responded with a `clean_sync` in the preflight stage then it is expected that the client will throw out its rules and the sync server should supply the client with all of the rules configured for it. 
+
+If a clean sync was not requested by either the client or the sync service, then the sync service should only send new rules seen since the last time the client synced.
+
+#### `ruledownload` Request
+
+Rules are retrieved from the sync server by having the client (Santa) issues an
+HTTP POST request to the url `/ruledownload/<machine_id>`
+
+| Key | Required | Type | Meaning |
+|---|---|---|---|
+| cursor | NO | string | a field used by the Sync Server to indicate where the next batch of rules should start |
+
+
+##### `ruledownload` Request Example Payload 
+
+On the first request the payload is an empty dictionary
+
+```json
+{}
+```
+
+The server will then send a response with the number of rules specified in the preflight response `rule_batch` the response will also contain a special field called `cursor`. The value and form of this field is left to the sync server implementor but is used to indicate that the client is ready for the next batch of rules.
+
+On subsequent requests to the server the cursor field is sent with the value from the previous response.
+
+```json
+{"cursor":"CpgBChcKCnVwZGF0ZWRfZHQSCQjh94a58uLlAhJ5ahVzfmdvb2dsZS5jb206YXBwbm90aHJyYAsSCUJsb2NrYWJsZSJAMTczOThkYWQzZDAxZGRmYzllMmEwYjBiMWQxYzQyMjY1OWM2ZjA3YmU1MmY3ZjQ1OTVmNDNlZjRhZWI5MGI4YQwLEgRSdWxlGICA8MvA0tIJDBgAIAA="}
+```
+
+#### `ruledownload` Response
+
+When a `ruledownload` request is received the sync server responds with a JSON object
+containing a list of rule objects and a cursor so the client can resume
+downloading if the rules need to be downloaded in multiple batches. 
+
+| Key | Required | Type | Meaning |
+|---|---|---|---|
+| cursor | NO | string | used to continue a rule download in a future request |
+| rules | YES | list of Rule objects | List of rule objects |
+
+##### Rules Objects
+
+
+:warning: Rule objects must have one of either the `identifier` or `sha256` fields.  The `sha256` field is supported for backwards compatability. 
```

Comment:
I'd simplify this by not mentioning the `sha256` field at all, we should consider it deprecated and only present for backward compatibility

Reply:
Removed.